app/models/topic_tracking_state.rb

```diff
@@ -89,8 +89,8 @@ def self.publish_latest(topic, staff_only = false)
     end
 
     group_ids =
-      if staff_only
-        [Group::AUTO_GROUPS[:staff]]
+      if whisper
+        SiteSetting.enable_whispers.split("|").map(&:to_i).presence
```

Comment:
I think we should just extra this logic into `site_setting.rb` instead of duplicating it everywhere.